boot/zephyr/main.c

```diff
@@ -60,11 +60,12 @@ void main(void)
 	}
 
 	printk("Bootloader chain: 0x%x\n", rsp.br_image_addr);
-	vt = (struct vector_table *)(rsp.br_image_addr +
-				     rsp.br_hdr->ih_hdr_size);
+	vt = (struct vector_table *)(rsp.br_image_addr + TEXT_SECTION_OFFSET);
```

Comment:
I wouldn't expect this to work. The bootloader should be built with a TEXT_SECTION_OFFSET of 0. Only the programs booted should themselves have it set.